src/sentry/buffer/redis.py

```diff
@@ -54,10 +55,11 @@ def flush(self):
 class RedisBuffer(Buffer):
     key_expire = 60 * 60  # 1 hour
     pending_key = 'b:p'
-    incr_batch_size = 2
 
-    def __init__(self, **options):
+    def __init__(self, pending_shards=1, incr_batch_size=2, **options):
         self.cluster, options = get_cluster_from_options('SENTRY_BUFFER_OPTIONS', options)
+        self.pending_shards = pending_shards
```

Comment:
`assert self.pending_shards > 0`?

Reply:
It's probably worth documenting somewhere that increasing the shard count is safe but decreasing the shard count can/will result in writes being lost.

Reply:
Yeah, I thought about doing this, but figured it'd just be us shooting ourselves in the foot. I'll follow up with some assertions to make sure we don't do that in a follow up so I don't wait another hour for CI.